Sources/NIOCore/EventLoopFuture.swift

```diff
@@ -2187,6 +2187,79 @@ extension EventLoopFuture {
     }
 }
 
+// MARK: assertion
+
+extension EventLoopFuture {
+    /// Returns a new `EventLoopFuture` that asserts the original future's success.
+    ///
+    /// If the original future fails, it triggers an assertion failure, causing a runtime error during development.
+    @inlinable
+    public func assertSuccess() -> EventLoopFuture<Value> {
+        let promise = self.eventLoop.makePromise(of: Value.self)
+        self.whenComplete { result in
+            switch result {
+            case .success(let value):
+                promise.succeed(value)
+            case .failure(let error):
+                assertionFailure("Expected success, but got failure: \(error)")
```

Comment:
Can we pass `file` and `line` through from the line that calls `assertSuccess`? That'll make the crash more generally useful.